cmd/zoekt-sourcegraph-indexserver/index.go

```diff
@@ -234,9 +251,16 @@ func gitIndex(c gitIndexConfig, o *indexArgs, sourcegraph Sourcegraph, l sglog.L
 			"-C", gitDir,
 			"-c", "protocol.version=2",
 			"-c", "http.extraHeader=X-Sourcegraph-Actor-UID: internal",
-			"fetch", "--depth=1", o.CloneURL,
+			"fetch", "--depth=1", "--no-tags",
 		}
 
+		// If there are no exceptions to MaxFileSize (1MB), we can avoid fetching these large files.
+		if len(o.LargeFiles) == 0 {
+			fetchArgs = append(fetchArgs, "--filter=blob:limit=1m")
```

Comment:
Here's the interesting part of the change.

Reply:
very cool! In the past I explored wanting to add a smarter way to fetch git bundles to handle things like the ignore rule. But given for most places this isn't set this is a great optimization.